rpc/execution_pool.go

```diff
@@ -33,11 +41,11 @@ func NewExecutionPool(initialSize int, timeout time.Duration) *SafePool {
 		return sp
 	}
 
-	var ptr atomic.Pointer[workerpool.WorkerPool]
+	sp.executionPool.Store(workerpool.New(initialSize))
 
-	p := workerpool.New(initialSize)
-	ptr.Store(p)
-	sp.executionPool = &ptr
+	if metrics.Enabled && report {
+		go sp.reportMetrics(3 * time.Second)
```

Comment:
What is the typical duration to collect metrics? Could we use the same value here?

Reply:
Yes those are sent every 3 seconds and hence I've kept the same here.